File: caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        // Convert entered key from string to integer type
        int key = atoi(argv[1]);

        string plain = get_string("Enter the code you want to encrypt:\n");

        printf("ciphertext: ");

        for (int n = 0; n < strlen(plain); n++)
        {
            // Check if plain character is a letter
            if (isalpha(plain[n]))
            {
                int ASCII = plain[n];

                // Process for uppercase character
                if (isupper(plain[n]))
                {
                    ASCII = ASCII - 65;
                    int aIndex = (ASCII + key) % 26;
                    ASCII = aIndex + 65;
                    printf("%c", ASCII);
                }

                // Process for lowercase character
                else if (islower(plain[n]))
                {
                    ASCII = ASCII - 97;
                    int aIndex = (ASCII + key) % 26;
                    ASCII = aIndex + 97;
                    printf("%c", ASCII);
                }

                // Simply print as character is if not a letter
            }
            else
            {
                printf("%c", plain[n]);
            }
        }
        printf("\n");
    }

    // If argc condition is not met
    else
    {
        return 1;
    }
}

File: caesar.py
import sys
import cs50


def main():
    if len(sys.argv) != 2:
        # Exit out of the program with '1' if there are not two command line arguments
        sys.exit(1)
    else:
        # Convert the key entered to an integer value and prompt what must be enxrypted
        key = int(sys.argv[1])
        plain = cs50.get_string("Enter the code you want to encrypt:\n")
        print("ciphertext: ", end="")

        for x in plain:
            # Check for whether the character is alphabetical
            if x.isalpha() == True:
                # Convert letter to its ASCII value
                ASCII = ord(x)
                # Check if character is uppercase or lowercase
                if x.isupper() == True:
                    ASCII = ASCII - 65
                    aIndex = (ASCII + key) % 26
                    ASCII = aIndex + 65
                    print(chr(ASCII), end="")
                elif x.islower() == True:
                    ASCII = ASCII - 97
                    aIndex = (ASCII + key) % 26
                    ASCII = aIndex + 97
                    print(chr(ASCII), end="")
            # If not alphabetical, print as it is
            else:
                print(x, end="")
        print()


if __name__ == "__main__":
    main()